src/components/Form/formFunctions.ts

```diff
@@ -47,25 +50,20 @@ export interface Fields {
     [fieldKey: string]: any;
 }
 
-export interface FormErrors {
-    [errorKey: string]: Errors;
-}
-
-export const setError = (
-    errorSetter: (errors: FormErrors) => void,
-    errorValues: FormErrors
-): ErrorSetter => {
-    return (errorKey, errorType) => {
-        errorSetter({ ...errorValues, [errorKey]: errorType });
-    };
+export type FormErrors<SpecificFields extends Fields> = {
+    [errorKey in keyof SpecificFields]?: Errors;
 };
 
-export const setField = <SpecificFields extends Fields>(
-    fieldSetter: (fieldValues: SpecificFields) => void,
+export const createSetter = <SpecificFields extends Fields>(
+    setFields: (SpecificFields: SpecificFields) => void,
     fieldValues: SpecificFields
-): FieldSetter => {
-    return (fieldKey, newFieldValue) => {
-        fieldSetter({ ...fieldValues, [fieldKey]: newFieldValue });
+): Setter<SpecificFields> => {
+    return <Key extends keyof SpecificFields>(keyValuePairs: [Key, any][]): void => {
+        const newFieldValues: SpecificFields = { ...fieldValues };
+        keyValuePairs.forEach((keyValuePair) => {
+            newFieldValues[keyValuePair[0]] = keyValuePair[1];
+        });
+        setFields(newFieldValues);
```

Comment:
This new createSetter function takes in a state and a setState action, and returns a setter function which takes in key value pairs and updates the corresponding keys in the state. The differences are that now the setter function can update multiple keys simultaneously, and also the createSetter can be called for both the fields and the error fields.